### PR TITLE
[ESIMD] Doxygen update VI - overall structure, misc clean-up, additions.

### PR DESCRIPTION
**Pull request by kbobrovs**
Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_mask_impl.hpp [hunk 1]
```diff
@@ -21,24 +21,42 @@ namespace experimental {
...
 
+/// @addtogroup sycl_esimd_core_vectors
+/// @{
+
+/// This class is a simd_obj_impl specialization representing a simd mask, which
+/// is basically a simd_obj_impl with fixed element type (uint16_t, not publicly
+/// usable) and limited set of APIs. E.g. arithmetic operations (\c +, \c -,
+/// etc.) are not defined for masks, but bit operations like \c ^, \c & are.
+/// @tparam T Fixed element type, must be simd_mask_elem_type.
+/// @tparam N Number of elements (per-lane predicates) in the mask.
```
[the remark this answers is not in the document]
Comment (kbobrovs) [reply sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_mask_impl.hpp:32]: Right, thanks for catching. There is one place where users need to know the distinction between least significant bit/non-zero - this is conversion from a simd object. Will add.

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_mask_impl.hpp [hunk 2]
```diff
@@ -25,9 +25,35 @@ namespace detail {
...
+/// @endcode
+///   In this and all other cases below lanes with non-zero value are treated as
+///   "enabled", lanes with zero - as "disabled".
+/// - constructing from an array or using broadcast constructor:
+/// @code{.cpp}
+///   simd_mask<8> m1({1,0,1,0,1,0,1,0});
+///   simd_mask<8> m2(1); // all "enabled"
+/// @endcode
+/// - constructing from a @ref simd object.
+/// Use code should use <code>simd_mask<1>::element_type<code> when the mask
```
Comment (kychendev) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_mask_impl.hpp:53]: Typo: Use code -> User code

File: sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_mask_impl.hpp [hunk 3]
```diff
@@ -21,24 +21,68 @@ namespace experimental {
...
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/detail/simd_mask_impl.hpp:53]:
```suggestion
/// User code should use <code>simd_mask<1>::element_type<code> when the mask
```

File: sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp [hunk 4]
```diff
@@ -20,34 +20,44 @@ namespace intel {
...
+/// @addtogroup sycl_esimd_core_vectors
 /// @{
 
 /// This class represents a reference to a sub-region of a base simd object.
 /// The referenced sub-region of the base object can be read from and written to
-/// via an instance of this class.
+/// via an instance of this class. Derives from detail::simd_view_impl, which
+/// defines the majority of available APIs for view classes.
+///
+/// Use code is never supposed to explicitly provide actual template parameters
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp:31]:
```suggestion
/// User code is never supposed to explicitly provide actual template parameters
```

**Review by kychendev — APPROVED** [review]

**Pull request merged**